sdk/samples/keyvault/keyvault/samples/src/keys_client_example.c

```diff
@@ -125,36 +125,44 @@ int main()
       &http_response);
 
   // validate sample running with no_op http client
-  if (create_result == AZ_ERROR_NOT_IMPLEMENTED)
+  if (key_create_result == AZ_ERROR_NOT_IMPLEMENTED)
   {
     printf("Running sample with no_op HTTP implementation.\nRecompile az_core with an HTTP client "
            "implementation like CURL to see sample sending network requests.\n\n"
            "i.e. cmake -DTRANSPORT_CURL=ON ..\n\n");
-    return 0;
+    return 4;
   }
 
-  TEST_FAIL_ON_ERROR(create_result, "Failed to create key");
+  if (az_failed(key_create_result))
+  {
+    printf("\nFailed to create key\n");
+    return 5;
+  }
 
   printf("Key created result: \n%s", response_buffer);
 
   /****************** 6) GET KEY latest ver ******************************/
-  az_result get_key_result = az_keyvault_keys_key_get(
+  az_result key_get_result = az_keyvault_keys_key_get(
       &client, &az_context_app, key_name_for_test, AZ_SPAN_NULL, &http_response);
 
-  TEST_FAIL_ON_ERROR(get_key_result, "Failed to get key");
+  if (az_failed(key_get_result))
+  {
+    printf("\nFailed to get key\n");
+    return 6;
+  }
 
   printf("\n\n*********************************\nGet Key result: \n%s", response_buffer);
 
   /****************** 7) get key version from response ****/
   // version is still at http_response. Let's copy it to a new buffer
-  uint8_t copy_version_buf[40];
-  az_span copy_version_builder = AZ_SPAN_FROM_BUFFER(copy_version_buf);
+  uint8_t copy_version_buf[40] = { 0 };
```

Comment:
nit: maybe naming the array and the span distinctly would be clearer.
```suggestion
  uint8_t copy_version_buffer[40] = { 0 };
```